File: app/src/main/java/net/mbonnin/arcanetracker/model/GameSummary.java
package net.mbonnin.arcanetracker.model;

import net.mbonnin.arcanetracker.PaperDb;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class GameSummary {
    public String deckName;
    public int hero;
    public int opponentHero;
    public boolean coin;
    public boolean win;
    public String date;

    public String hsreplayUrl;
    // added in v213
    public int bnetGameType;

    static final String KEY_GAME_LIST = "KEY_GAME_LIST";
    static List<GameSummary> savedGameSummaryList;

    static {
        savedGameSummaryList = PaperDb.INSTANCE.read(KEY_GAME_LIST);
        if (savedGameSummaryList == null) {
            savedGameSummaryList = new ArrayList();
        }
    }

    static public void eraseGameSummary() {
        savedGameSummaryList.clear();
        sync();
    }

    public static void addFirst(@NotNull GameSummary summary) {
        savedGameSummaryList.add(0, summary);
        sync();
    }

    public static void sync() {
        PaperDb.INSTANCE.write(KEY_GAME_LIST, savedGameSummaryList);
    }

    public static List<GameSummary> getGameSummary() {
        return savedGameSummaryList;
    }
}

File: app/src/main/kotlin/net/mbonnin/arcanetracker/model/GameSummary.kt
package net.mbonnin.arcanetracker.model

import net.mbonnin.arcanetracker.PaperDb
import java.util.*

class GameSummary {
    var deckName: String? = null
    var hero: Int = 0
    var opponentHero: Int = 0
    var coin: Boolean = false
    var win: Boolean = false
    var date: String? = null

    var hsreplayUrl: String? = null
    // added in v213
    var bnetGameType: Int = 0

    companion object {

        internal val KEY_GAME_LIST = "KEY_GAME_LIST"
        internal var savedGameSummaryList: ArrayList<GameSummary>? = null

        init {
            savedGameSummaryList = PaperDb.read<ArrayList<GameSummary>?>(KEY_GAME_LIST)?.let { ArrayList(it) }
            if (savedGameSummaryList == null) {
                savedGameSummaryList = ArrayList()
            }
        }

        val gameSummaryList: List<GameSummary>?
            get() = savedGameSummaryList

        fun sync() {
            PaperDb.write<ArrayList<GameSummary>?>(KEY_GAME_LIST, savedGameSummaryList)
        }

        fun eraseGameSummary() {
            savedGameSummaryList!!.clear()
            sync()
        }
    }
}
